src/linear_algebra/nonsingular_inverse.lean

```diff
@@ -0,0 +1,300 @@
+/-
+  Copyright (c) 2019 Tim Baanen. All rights reserved.
+  Released under Apache 2.0 license as described in the file LICENSE.
+  Author: Tim Baanen.
+
+  Inverses for nonsingular square matrices.
+-/
+import algebra.big_operators
+import data.matrix.basic
+import linear_algebra.determinant
+
+/-!
+# Nonsingular inverses
+
+In this file, we define an inverse for square matrices of invertible
+determinant. For matrices that are not square or not of full rank, there is a
+more general notion of pseudoinverses. Unfortunately, the definition of
+pseudoinverses is typically in terms of inverses of nonsingular matrices, so we
+need to define those first. The file also doesn't define a `has_inv` instance
+for `matrix` so that can be used for the pseudoinverse instead.
+
+The definition of inverse used in this file is the one given by Cramer's rule.
```

Comment:
It would help to give the definition of `cramer` and `adjugate` in the docstring, as these are not completely standard notions.